src/ddmd/expression.h

```diff
@@ -773,6 +773,8 @@ class DotIdExp : public UnaExp
 {
 public:
     Identifier *ident;
+    bool noderef;       // true if the result of the expression will never be dereferenced
```

Comment:
The header file needs to have the same type declarations as the .d file, for compatibility with C++. `noderef` was mistakenly missing here. Adding it here now was required when adding a new field to the .d file, as otherwise they wouldn't match up.